x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/frequentitemsets/mr/ItemSetMapReduceAggregator.java

```diff
@@ -126,14 +126,19 @@ protected LeafBucketCollector getLeafCollector(AggregationExecutionContext ctx,
             )
             : null;
 
+        List<ItemSetMapReduceValueSource.ValueCollector> valueCollectors = new ArrayList<>();
```

Comment:
Is it worth providing the size of the list (which we know in advance) upon creation?